dask/dataframe/io/parquet/arrow.py

```diff
@@ -698,10 +300,476 @@ def read_metadata(
+
+        index_cols = index or ()
+        meta = _meta_from_dtypes(all_columns, dtypes, index_cols, column_index_names)
+        meta = clear_known_categories(meta, cols=categories)
+
+        if partition_obj:
+
+            for partition in partition_obj:
+                if isinstance(index, list) and partition.name == index[0]:
+                    # Index from directory structure
+                    meta.index = pd.CategoricalIndex(
+                        categories=partition.keys, name=index[0]
+                    )
+                elif partition.name == meta.index.name:
+                    # Index created from a categorical column
+                    meta.index = pd.CategoricalIndex(
+                        categories=partition.keys, name=meta.index.name
+                    )
+                elif partition.name in meta.columns:
+                    meta[partition.name] = pd.Series(
+                        pd.Categorical(categories=partition.keys, values=[]),
+                        index=meta.index,
+                    )
+
+        return meta, index_cols, categories, index
+
+    @classmethod
+    def _construct_parts(
+        cls,
+        fs,
+        metadata,
+        schema,
+        filters,
+        index_cols,
+        data_path,
+        partition_info,
+        categories,
+        split_row_groups,
+        gather_statistics,
+        fragment_row_groups,
+        use_legacy_dataset=False,
+    ):
+        """Construct ``parts`` for ddf construction
+
+        Use metadata (along with other data) to define a tuple
+        for each ddf partition.  Also gather statistics if
+        ``gather_statistics=True``, and other criteria is met.
+        """
+
+        parts = []
+        stats = []
+
+        partition_keys = partition_info["partition_keys"]
+        partition_obj = partition_info["partitions"]
+
+        # Check if `metadata` is just a list of paths
+        # (not splitting by row-group or collecting statistics)
+        if (
+            isinstance(metadata, list)
+            and len(metadata)
+            and isinstance(metadata[0], str)
+        ):
+            for full_path in metadata:
+                part = {
+                    "piece": (full_path, None, partition_keys.get(full_path, None)),
+                    "kwargs": {"partitions": partition_obj, "categories": categories},
+                }
+                parts.append(part)
+            return parts, stats
+
+        # Determine which columns need statistics
+        flat_filters = (
+            set(flatten(tuple(flatten(filters, container=list)), container=tuple))
+            if filters
+            else []
+        )
+        stat_col_indices = {}
+        for i, name in enumerate(schema.names):
+            if name in index_cols or name in flat_filters:
+                stat_col_indices[name] = i
+        stat_cols = list(stat_col_indices.keys())
+        gather_statistics = gather_statistics and len(stat_cols) > 0
```

Comment:
Should this also take into account if a filter is about a partition column? (for which there are no statistics to gather)

Reply:
Ah - Good point.  The partition columns will not have statistics.